transforms/universal/fdedup/python/src/cluster_analysis_transform.py

```diff
@@ -0,0 +1,336 @@
+        doc_list = [item["int_id_column"] for item in sorted_document_data]
+
+        # Creating a dictionary with int_id_column as key and minhashes as value
+        doc_minhashes = {item["int_id_column"]: item["minhashes"] for item in sorted_document_data}
+
+        while len(doc_list) > 1:
+            docs_to_remove = []
+            new_doc_list = []
+            # this is the document we are going to keep
+            first_doc = doc_list[0]
+            first_mh = doc_minhashes[first_doc]
+            for int_id_column in doc_list[1:]:
+                doc_mh = doc_minhashes[int_id_column]
+                distance = Murmur_MH.jaccard(np.array(first_mh), np.array(doc_mh))
+                if distance >= threshold:
+                    docs_to_remove.append(int_id_column)
+                else:
+                    new_doc_list.append(int_id_column)
+            if len(docs_to_remove) > 0:
+                docs_to_remove = list(set(docs_to_remove))
+                doc_ids_list.append(first_doc)
+                docs_to_remove_list.append(docs_to_remove)
+                len_of_docs2remove_list.append(len(docs_to_remove))
+            doc_list = new_doc_list
+
+        return doc_ids_list, docs_to_remove_list, len_of_docs2remove_list
+
+
+class ClusterAnalysisTransformConfiguration(TransformConfiguration):
+
+    """
+    Provides support for configuring and using the associated Transform class include
+    configuration with CLI args.
+    """
+
+    def __init__(self):
+        super().__init__(
+            name=short_name,
+            transform_class=ClusterAnalysisTransform,
+            remove_from_metadata=[],
+        )
+        self.logger = get_logger(__name__, level="INFO")
+
+    def add_input_params(self, parser: ArgumentParser) -> None:
+        """
+        Add Transform-specific arguments to the given  parser.
+        This will be included in a dictionary used to initialize the NOOPTransform.
+        By convention a common prefix should be used for all transform-specific CLI args
+        (e.g, noop_, pii_, etc.)
+        """
+        parser.add_argument(
+            f"--{jaccard_similarity_threshold_cli_param}",
+            type=float,
+            default=jaccard_similarity_threshold_default,
+            help="Jaccard similarity threshold above which two documents are duplicates",
+        )
+        parser.add_argument(
+            f"--{num_bands_cli_param}",
+            type=int,
+            default=num_bands_default,
+            help="The number of bands used in the banding technique",
+        )
+        parser.add_argument(
+            f"--{num_segments_cli_param}",
+            type=int,
+            default=num_segments_default,
+            help="The number of segments dividing the hashing space for each band",
+        )
+        parser.add_argument(
+            f"--{sort_output_cli_param}",
+            type=bool,
+            default=sort_output_default,
+            help="Sort",
```

Comment:
sort all output or within a file and sort by what?

Reply:
Fixed in commit https://github.com/IBM/data-prep-kit/pull/699/commits/8de14b51e55a69bfa7f4dddc763500fb3a01db4d